node/src/components/rpc_server/event.rs

```diff
@@ -23,8 +23,8 @@ pub enum Event {
     RpcRequest(RpcRequest<NodeId>),
     GetBlockResult {
         maybe_id: Option<BlockIdentifier>,
-        result: Box<Option<Block>>,
-        main_responder: Responder<Option<Block>>,
```

Comment:
No it can remain as `GetBlockResult` as from the perspective of the RPC event, it really cares about retrieving a block to send to external callers which includes the signatures as metadata. Its also the convention we have with `Deploy` even though there is associated metadata there too